modules/memberships/class-jetpack-memberships.php

```diff
@@ -250,16 +250,19 @@ public function render_button( $attrs ) {
 		}
 		$button_styles = implode( ';', $button_styles );
 		add_thickbox();
+		global $wp;
 		return sprintf(
-			'<div class="wp-block-button %1$s"><a role="button" data-blog-id="%2$d" data-powered-text="%3$s" data-plan-id="%4$d" data-lang="%5$s" class="%6$s" style="%7$s">%8$s</a></div>',
+			'<div class="wp-block-button %1$s"><a role="button" %10$s href="https://subscribe.wordpress.com/memberships/?blog=%2$d&plan=%4$d&lang=%5$s&pid=%9$d&redirect=%3$s" class="%6$s" style="%7$s">%8$s</a></div>',
```

Comment:
Since that URL is getting quite long, I wonder if it wouldn't be more readable to build it beforehand, like this for example?

```diff
diff --git a/modules/memberships/class-jetpack-memberships.php b/modules/memberships/class-jetpack-memberships.php
index 798861aac..d5dc4810f 100644
--- a/modules/memberships/class-jetpack-memberships.php
+++ b/modules/memberships/class-jetpack-memberships.php
@@ -250,18 +250,33 @@ class Jetpack_Memberships {
 		}
 		$button_styles = implode( ';', $button_styles );
 		add_thickbox();
+
 		global $wp;
+
+		$button_url = add_query_arg(
+			array(
+				'blog'     => esc_attr( $data['blog_id'] ),
+				'plan'     => esc_attr( $data['id'] ),
+				'lang'     => esc_attr( get_locale() ),
+				'pid'      => esc_attr( get_the_ID() ), // Needed for analytics purposes.
+				'redirect' => esc_attr( rawurlencode( home_url( $wp->request ) ) ), // Needed for redirect back in case of redirect-based flow.
+			),
+			'https://subscribe.wordpress.com/memberships/'
+		);
+
 		return sprintf(
-			'<div class="wp-block-button %1$s"><a role="button" %6$s href="https://subscribe.wordpress.com/memberships/?blog=%2$d&plan=%4$d&lang=%5$s&pid=%9$d&redirect=%3$s" class="%6$s" style="%7$s">%8$s</a></div>',
-			esc_attr( Jetpack_Gutenberg::block_classes( self::$button_block_name, $attrs ) ),
-			esc_attr( $data['blog_id'] ),
-			esc_attr( rawurlencode( home_url( $wp->request ) ) ), // Needed for redirect back in case of redirect-based flow.
-			esc_attr( $data['id'] ),
-			esc_attr( get_locale() ),
+			'<div class="%1$s"><a role="button" %10$s href="%2$s" class="%3$s" style="%4$s">%5$s</a></div>',
+			esc_attr(
+				Jetpack_Gutenberg::block_classes(
+					self::$button_block_name,
+					$attrs,
+					array( 'wp-block-button' )
+				)
+			),
+			esc_url( $button_url ),
 			isset( $attrs['submitButtonClasses'] ) ? esc_attr( $attrs['submitButtonClasses'] ) : 'wp-block-button__link',
 			esc_attr( $button_styles ),
 			wp_kses( $data['button_label'], self::$tags_allowed_in_the_button ),
-			esc_attr( get_the_ID() ), // Needed for analytics purposes.
 			isset( $attrs['submitButtonAttributes'] ) ? sanitize_text_field( $attrs['submitButtonAttributes'] ) : '' // Needed for arbitrary target=_blank on WPCOM VIP.
 		);
 	}

```

Reply:
Good idea, done!